### PR TITLE
Bug/fix over permissive actions

### PR DESCRIPTION
**Pull request by knastase**
Pull request

## What does this PR do?

[//]: # (Required: Describe the effects of your pull request in detail. If
multiple changes are involved, a bulleted list is often useful.)
Adds the ability to toggle whether to include dependent actions as wildcards *if they don't share the same resource type. I had a hard time coming up with how to explain this in the crud template via a comment. We can change the wording if necessary.

I focused on addressing two specific situations I found, but there could be more this would fix.
1. S3 write on a bucket ARN would include `iam:PassRole -> Resource: *` giving the user of this policy the ability to pass any role to a service.
2. KMS write on a key ARN would incude `kms:PutKeyPolicy -> Resource: *` giving the user of this policy the ability to modify any key policy *if a default or overly permissive key policy is being used.

One thing I noticed while testing is, even if we add the action that has dependent actions to `exclude-actions` PS will still include its dependent actions. Maybe a bug?

TL;DR; In addition to adding a toggle to add dependent actions as wildcard it also moves dependent actions with the same resource into the same SID (instead of wildcard).

## What gif best describes this PR or how it makes you feel?

[//]: # (Encouraged: Insert an appropriate gif/meme to brighten up your reviewer's day.)
image

## Completion checklist

- [ ] Additions and changes have unit tests
- [ ] [Unit tests, Pylint, security testing, and Integration tests are passing.](https://github.com/salesforce/policy_sentry/actions). GitHub actions does this automatically
- [ ] The pull request has been appropriately labeled using the provided PR labels

### CODE REVIEW

**Comment by kmcquade** [comment]
@knasty51 I really like this PR! Nice job.

I noticed that you set `wildcard-dependent-actions` to `True` by default. What do you think about setting it to `False` by default? I understand backwards compatibility, but I'm all for creating a new minor version (where some breaking changes are acceptable), flagging it in the release notes. Since doing it otherwise would be a security issue, I think.

> One thing I noticed while testing is, even if we add the action that has dependent actions to exclude-actions PS will still include its dependent actions. Maybe a bug?

Yes that definitely sounds like a bug. I can approve this PR and then we can open a separate bug on that. What do you think?

**Comment by knasty51** [comment]
Hey @kmcquade thanks!!!

I'm really glad you brought this up because I was having an internal conflict. The developer in me says "backwards compatibility" but the security side says "Give less permissions" or rather don't throw extra actions in unless I am expecting it. I was playing it safe as I did not want to alter the current functionality if users (even by mistake) are relying on those dependent actions being in the policy document (or maybe they found them and added them to the exclude list already). I am fine with either but I was also leaning towards false myself if this was an initial commit to the project. We can certainly reverse the logic, but I fear for current users who don't read release notes.

If we go down that route, it brings up the question of the template key wording - whether we should change it to something like `include-other-service-dependent-actions-as-wildcard: true` (ok that's verbose but I hope you get the picture) or abstracting the functionality entirely by not including it in the default template unless the user knows what they're doing or reading the documentation. In that case, they might just add it as a single action if they really need it. I foresee adding dependent actions within the same service/resource constraint (as this PR does) to be enough for most use cases. 

My aim is to not muck up the crud template or make it so verbose that they might as well write CloudFormation YAML to produce a policy they really want. Trying to find that delicate balance. I appreciate your input! 

P.S. I will test to double check that is really a bug I found and then open an issue for it.
